lib/qesapdeployment.pm

```diff
@@ -2029,4 +2031,83 @@ sub qesap_az_create_sas_token {
     record_info('GENERATE-SAS', $cmd);
     return script_output($cmd);
 }
+
+=head2 qesap_terraform_clean_up_retry
+
+    qesap_terraform_clean_up_retry()
+
+    Do terraform cleanup only when ansible failed on 'Timed out waiting for last boot time check'
```

Comment:
> Do terraform cleanup only when ansible failed on 'Timed out waiting for last boot time check'

There's nothing in the function implementation about `'Timed out waiting for last boot time check'` .  There's nothing that prevent someone to call this function in other situations too.

What about documenting it like this:
```
Perform terraform destroy and catch and ignore any error. This method is mostly useful when doing cleanup before retry in case of Ansible failed on 'Timed out waiting for last boot time check'
```

Reply:
Revised comments as suggested: `"erform terraform destroy, catch ...`